server/vmap/trillian_map_server.go

```diff
@@ -102,19 +102,30 @@ func (t *TrillianMapServer) GetLeaves(ctx context.Context, req *trillian.GetMapL
 		KeyValue: make([]*trillian.KeyValueInclusion, 0, len(req.Key)),
 	}
 
+	keyHashes := make([]trillian.Hash, 0, len(req.Key))
+	hashToKey := make(map[string][]byte)
 	for _, key := range req.Key {
 		kHash := kh.HashKey(key)
+		keyHashes = append(keyHashes, kHash)
+		hashToKey[string(kHash)] = key
+	}
+
+	leaves, err := tx.Get(req.Revision, keyHashes)
```

Comment:
No, it'll return what it had - the caller must use the `KeyHash` entry in the leaf struct to determine the association between key/value.